e2e/controls.py

```diff
@@ -2336,3 +2336,188 @@ def expect_n_row(
             n,
             timeout=timeout,
         )
+
+
+class _CardBodyP(_InputBaseP, Protocol):
+    loc: Locator
```

Comment:
I don't mind having a class to give it a clearer name, but `loc` is redundant as it's in `_InputBaseP`.

```suggestion
```